developer-hub/src/polkadot_sdk/frame_runtime.rs

```diff
@@ -0,0 +1,129 @@
+//!
+//! - [Config](frame::pallet_macros::config), allowing a pallet to make itself configurable and
+//!   generic over types, values and such.
+//! - [Storage](frame::pallet_macros::storage), allowing a pallet to define onchain storage.
+//! - [Dispatchable function aka. Extrinsics](frame::pallet_macros::call), allowing a pallet to
+//!   define extrinsics that are callable by end users, from the outer world.
+//! - [Events](frame::pallet_macros::event), allowing a pallet to emit events.
+//! - [Errors](frame::pallet_macros::error), allowing a pallet to emit well-formed errors.
+//!
+//! Most of these components are defined using macros, the full list of which can be found in
+//! [`frame::pallet_macros`]
+//!
+//! ### Example
+//!
+//! The following examples showcases a minimal pallet.
+#![doc = docify::embed!("src/polkadot_sdk/frame_runtime.rs", pallet)]
+//!
+//! ## Runtime
+//!
+//! A runtime is a collection of pallets that are amalgamated together. Each pallet typically has
+//! some configurations (exposed as a `trait Config`) that needs to be *specified* in the runtime.
+//! This is done with [`frame::runtime::prelude::construct_runtime`].
+//!
+//! A (real) runtime that actually wishes to compile to WASM needs to also implement a set of
+//! runtime-apis. These implementation can be specified using the
+//! [`frame::runtime::prelude::impl_runtime_apis`] macro.
+//!
+//! ### Example
+//!
+//! The following example shows a (test) runtime that is composing the pallet demonstrated above,
+//! next to the [`frame::prelude::frame_system`] pallet, into a runtime.
+#![doc = docify::embed!("src/polkadot_sdk/frame_runtime.rs", runtime)]
+//!
+//! ## More Examples
+//!
+//! You can find more FRAME examples that revolve around specific features at [`pallet_examples`]
+
+#[cfg(test)]
+mod tests {
+	use frame::prelude::*;
+
+	#[docify::export]
+	#[frame::pallet(dev_mode)]
+	pub mod pallet {
+		use super::*;
+
+		#[pallet::config]
+		pub trait Config: frame_system::Config {
+			type RuntimeEvent: IsType<<Self as frame_system::Config>::RuntimeEvent>
+				+ From<Event<Self>>;
+		}
+
+		#[pallet::pallet]
+		pub struct Pallet<T>(PhantomData<T>);
+
+		#[pallet::event]
+		pub enum Event<T: Config> {}
+
+		#[pallet::storage]
+		pub type Value<T> = StorageValue<Value = u32>;
+
+		#[pallet::call]
+		impl<T: Config> Pallet<T> {
+			pub fn some_dispatchable(_origin: OriginFor<T>) -> DispatchResult {
+				Ok(())
+			}
+		}
+	}
+
+	#[docify::export]
+	pub mod runtime {
+		use super::pallet as pallet_example;
+		use frame::{prelude::*, testing_prelude::*};
+
+		construct_runtime!(
+			pub struct Runtime {
+				System: frame_system,
+				Example: pallet_example,
+			}
+		);
+
+		#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
+		impl frame_system::Config for Runtime {
+			type Block = MockBlock<Self>;
+		}
+
+		impl pallet_example::Config for Runtime {
+			type RuntimeEvent = RuntimeEvent;
+		}
+	}
+}
```

Comment:
Would it make sense to add some inline comments describing what each part is doing?

Reply:
Actually yeah, one of my self criticisms so far is that most of the code that I use with docify is totally undocumented. It should be improved.

Reply:
Much better documented now.